### ISSUE TITLE
Allow search for all data sets of a component

### ISSUE THREAD

**Issue by christoph-maurer** [building-envelope-data/metabase#142]
When I query https://www.buildingenvelopedata.org/graphql/ with
```
query {
  databases {
    nodes {
      uuid
      name
      locator
      allData(
        where: {
          componentId: { 
equalTo: "34f7aebd-5d11-4db7-a5d0-18b51af8f817" }
        }
      ) {
        nodes {
          resourceTree {
            root {
              value {
                locator
                dataFormatId
              }
            }
          }
        }
      }
    }
  }
}
```
I get the result
```
{
  "errors": [
    {
      "message": "Unexpected Execution Error",
      "locations": [
        {
          "line": 7,
          "column": 7
        }
      ],
      "path": [
        "databases",
        "nodes",
        2,
        "allData"
      ]
    }
  ],
  "data": {
    "databases": {
      "nodes": [
        {
          "uuid": "e744cd25-a4bb-4d48-87e6-81f47ab49c3b",
          "name": "Daylighting Database",
          "locator": "http://daylightingdatabase.herokuapp.com/graphql/",
          "allData": null
        },
        {
          "uuid": "48994b60-670d-488d-aaf7-53333a64f1d6",
          "name": "IGSDB",
          "locator": "https://igsdb-icon.herokuapp.com/icon_graphql/",
          "allData": null
        },
        {
          "uuid": "8a27aa0d-6026-4124-b185-4efd5cead953",
          "name": "TestLab DB",
          "locator": "https://www.solarbuildingenvelopes.com/graphql/",
          "allData": null
        }
      ]
    }
  }
}
```
How can I query for all data sets of a component? It was possible in an earlier version of the metabase.

**Comment by christoph-maurer** [comment]
@simon-wacker Should the IGSDB change anything to facilitate a search for all data of all components at the website of the metabase?

**Comment by simon-wacker** [comment]
@christoph-maurer Clicking `Filter` on [https://www.buildingenvelopedata.org/data](https://www.buildingenvelopedata.org/data) now returns all data of [https://www.solarbuildingenvelopes.com](https://www.solarbuildingenvelopes.com). No data from the IGSDB though because it returns `null`. Even the minimal query

```
query AllData {
  allData(where: {}) {
    nodes {
      uuid
    }
  }
}
```

returns `null` for `allData`. The response of this query is

```
{
  "data": null,
  "errors": [
    {
      "message": "Cannot return null for non-nullable field Query.allData.",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "allData"
      ]
    }
  ]
}
```

The actual query that https://www.buildingenvelopedata.org/data sends to the [IGSDB GraphQL endpoint](https://igsdb-icon.herokuapp.com/icon_graphql/) is

```
fragment DataFields on Data {
  uuid
  timestamp
  componentId
  name
  description
  warnings
  creatorId
  createdAt
  appliedMethod {
    methodId
    arguments {
      name
      value
    }
    sources {
      name
      value {
        dataId
        dataTimestamp
        dataKind
        databaseId
      }
    }
  }
  # approvals {
  #   timestamp
  #   signature
  #   keyFingerprint
  #   query
  #   response
  #   approverId
  # }
  resources {
    description
    hashValue
    locator
    dataFormatId
    archivedFilesMetaInformation {
      path
      dataFormatId
    }
  }
  resourceTree {
    root {
      vertexId
      value {
        description
        hashValue
        locator
        dataFormatId
        archivedFilesMetaInformation {
          path
          dataFormatId
        }
      }
    }
    nonRootVertices {
      vertexId
      parentId
      value {
        description
        hashValue
        locator
        dataFormatId
        archivedFilesMetaInformation {
          path
          dataFormatId
        }
      }
      appliedConversionMethod {
        methodId
        arguments {
          name
          value
        }
        sourceName
      }
    }
  }
  # approval {
  #   signature
  #   keyFingerprint
  # }
  locale
}

query AllData(
  $where: DataPropositionInput!
  $timestamp: DateTime
  $locale: Locale
) {
  allData(where: $where, timestamp: $timestamp, locale: $locale) {
    edges {
      # cursor
      node {
        __typename
        ...DataFields
      }
    }
    nodes {
      __typename
      ...DataFields
    }
    # totalCount
    # pageInfo {
    #   ...PageInfoFields
    # }
    # timestamp
  }
}
```

with the query variables

```
{
  "where": {},
  "timestamp": null,
  "locale": null
}
```

and the response is

```
{
  "data": null,
  "errors": [
    {
      "message": "Cannot return null for non-nullable field Query.allData.",
      "locations": [
        {
          "line": 77,
          "column": 3
        }
      ],
      "path": [
        "allData"
      ]
    }
  ]
}
```

**Comment by christoph-maurer** [comment]
When I query IGSDB Version 2 https://igsdb-v2.herokuapp.com/graphql/
```
query AllData {
  allData(where: {}) {
    edges {
      node {
        componentId
        resourceTree {
          root {
            value {
              locator
            }
          }
        }
      }
    }
  }
}
```
The result is correct. LBNL doesn't need to do anything. ISE needs to switch from IGSDB Version 1 to Version 2.

**Comment by simon-wacker** [comment]
In version 2, the IGSDB uses different type names than specified in [`database.graphql`](https://github.com/building-envelope-data/api/blob/develop/apis/database.graphql). For example, the query `allData` in the IGSDB has [`ProductTypeConnection`](https://igsdb-v2.herokuapp.com/graphql/) as return type, whereas it is [`DataConnection`](https://github.com/building-envelope-data/api/blob/develop/apis/database.graphql#L298-L313) in `database.graphql`. This is no problem, when types are not used in queries as in your example above. However, the actual query mentioned in https://github.com/building-envelope-data/metabase/issues/142#issuecomment-1420802604 uses the fragment `fragment DataFields on Data { ... }`. This fragment does not work on the IGSDB, because the type `Data` does not exist there. The corresponding type has the name `ProductType`.

**Comment by christoph-maurer** [comment]
#181 is closed, #182 contains all open topics.
